client/ayon_deadline/plugins/publish/global/collect_jobinfo.py

```diff
@@ -0,0 +1,240 @@
+# -*- coding: utf-8 -*-
+from collections import OrderedDict
+
+import ayon_api
+import pyblish.api
+from ayon_core.lib import (
+    BoolDef,
+    NumberDef,
+    TextDef,
+    EnumDef,
+    is_in_tests,
+    UISeparatorDef
+)
+from ayon_core.pipeline.publish import AYONPyblishPluginMixin
+from ayon_core.settings import get_project_settings
+from ayon_core.lib.profiles_filtering import filter_profiles
+
+from ayon_deadline.lib import FARM_FAMILIES, DeadlineJobInfo
+
+
+class CollectJobInfo(pyblish.api.InstancePlugin, AYONPyblishPluginMixin):
+    """Collect variables that belong to Deadline's JobInfo.
+
+    Variables like:
+    - department
+    - priority
+    - chunk size
+
+    """
+
+    order = pyblish.api.CollectorOrder + 0.420
+    label = "Collect Deadline JobInfo"
+
+    families = FARM_FAMILIES
+    targets = ["local"]
+
+    def process(self, instance):
+        attr_values = self._get_jobinfo_defaults(instance)
+
+        attr_values.update(self.get_attr_values_from_data(instance.data))
+        # do not set empty strings
+        attr_values = {
+            key: value
+            for key,value in attr_values.items()
+            if value != ""
+        }
+        job_info = DeadlineJobInfo.from_dict(attr_values)
+        instance.data["deadline"]["job_info"] = job_info
+
+    @classmethod
+    def get_attr_defs_for_instance(cls, create_context, instance):
+        if not cls.instance_matches_plugin_families(instance):
+            return []
+
+        if not instance["active"]:  # TODO origin_data seem not right
+            return []
+
+        # will be reworked when CreateContext contains settings and task types
+        project_name = create_context.project_name
+        project_settings = get_project_settings(project_name)
+
+        host_name = create_context.host_name
+
+        task_name = instance["task"]
+        folder_path = instance["folderPath"]
+        folder_entity = ayon_api.get_folder_by_path(project_name,folder_path)
+        task_entity = ayon_api.get_task_by_name(
+            project_name, folder_entity["id"], task_name)
```

Comment:
Note: This happens only to get task type. Fetching this, for each instance, is very slow. We should have option to get task type of an instance from create context -> @kalisp could you create issue?
```suggestion
        task_name = instance["task"]
        folder_path = instance["folderPath"]
        folder_entity = ayon_api.get_folder_by_path(project_name,folder_path)
        task_entity = ayon_api.get_task_by_name(
            project_name, folder_entity["id"], task_name)
```

Reply:
Replaced by new implementation from https://github.com/ynput/ayon-core/pull/967